p2p/tests/buckets.rs

```diff
@@ -12,8 +12,11 @@ mod ice {
 
     #[test]
     fn test_can_ice_an_address() {
-        let mut peers = Peers::new();
-        peers.set_ice_period(Duration::from_secs(1000));
+        let ice_period = Duration::from_secs(1000);
+        let mut peers = Peers {
```

Comment:
I suspect mutability is not needed here anymore. Let's see what clippy has to say.

Reply:
It was needed indeed because the `contains` method does write-back when the ice "melts".